apps/browser/app/page.tsx

```diff
@@ -90,7 +93,16 @@ function Dojo({ params }: { params: { id?: string } }) {
 
     if (!goalChatId) {
       goalChatId = await handleCreateNewChat();
-      await updateChatTitle({ chatId: goalChatId, title: goal })
+    }
+
+    const currentChat = chats?.find((c) => c.id === goalChatId);
```

Comment:
Ideally we'd await for `chats` to be defined, so that we're not trying to update a chat that may already have a name (but we just haven't finished fetching it yet).